WordPress/UITestsFoundation/XCTestCase+Utils.swift

```diff
@@ -0,0 +1,34 @@
+import XCTest
+
+public extension XCTestCase {
+
+    func removeApp(_ app: XCUIApplication = XCUIApplication()) {
+        // We need to store the app name before calling `terminate()` if we want to delete it.
+        // Otherwise, we won't be able to access it to read its name as the test will fail with:
+        //
+        // > Failed to get matching snapshot: Application org.wordpress is not running
+        //
+        // Launch the app to access its name so we can deleted it from Springboard
+        switch app.state {
+        case .unknown, .notRunning:
+            app.launch()
+        case _:
+            break
+        }
```

Comment:
This adds a bit of delay to the process because it needs to launch the app... I think it's worth it, but am open to suggestions. 

For example, it should be possible to create a new `removeAppNamed(_ name:)` method that only uses SpringBoard and the app name, without calling `app.terminate()`. We could then always use that one under the assumption that if an app is deleted is also terminated.